### PR TITLE
p2p/net/conn: only reuseport if avail

### PR DESCRIPTION
**Pull request by jbenet**
This PR makes reuseport check whether the sockopt is available.
Older linux kernels (<=3.8, etc) don't have it. see
http://stackoverflow.com/questions/21454879/how-to-detect-so-reuseport-availibility-at-run-time

### CODE REVIEW

**Comment by whyrusleeping** [comment]
Wooo! i can run the daemon again!!

File: Godeps/_workspace/src/github.com/jbenet/go-reuseport/available_unix.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	} else {
+		atomic.StoreInt32(&c.check, 0)
+	}
+}
+
+// Available returns whether or not SO_REUSEPORT is available in the OS.
+// It does so by attepting to open a tcp listener, setting the option, and
+// checking ENOPROTOOPT on error. After checking, the decision is cached
+// for the rest of the process run.
+func available() bool {
```
Comment (whyrusleeping) [inline Godeps/_workspace/src/github.com/jbenet/go-reuseport/available_unix.go]: Couldnt (for linux distros) we just check the kernel version? I mean, this is a good way to be certain of whether or not it is available, but checking the version number would be simpler

**Comment by jbenet** [comment]
only failure is: https://build.protocol-dev.com/job/race/16704/console it seems to be a timing problem exacerbated by -race

File: Godeps/_workspace/src/github.com/jbenet/go-reuseport/available_unix.go (hunk 1 above)
Comment (jbenet) [inline Godeps/_workspace/src/github.com/jbenet/go-reuseport/available_unix.go]: > Couldnt (for linux distros) we just check the kernel version?

Opening a listener once per run is very cheap and it _makes sure_ reuseport works.

**Pull request merged**